toolkit/Makefile

```diff
@@ -168,14 +173,38 @@ else
       override PACKAGE_URL_LIST   += https://packages.microsoft.com/azurelinux/$(RELEASE_MAJOR_ID)/preview/base/debuginfo/$(build_arch)
       override PACKAGE_URL_LIST   += https://packages.microsoft.com/azurelinux/$(RELEASE_MAJOR_ID)/preview/ms-oss/$(build_arch)
       override SRPM_URL_LIST      += https://packages.microsoft.com/azurelinux/$(RELEASE_MAJOR_ID)/preview/base/srpms
-      ifneq ($(wildcard $(PREVIEW_REPO)),)
-         override REPO_LIST += $(PREVIEW_REPO)
+      # Ensure we can find the preview repo file.
+      ifneq ($(PREVIEW_REPO),)
+         ifneq ($(wildcard $(PREVIEW_REPO)),)
+            override REPO_LIST += $(PREVIEW_REPO)
+         else
+            $(warning )
+            $(warning ######################### ERROR #########################)
+            $(warning 'USE_PREVIEW_REPO=y' and 'PREVIEW_REPO=<path>' are set but)
+            $(warning '$(PREVIEW_REPO)' is missing.)
+            $(warning ######################### ERROR #########################)
+            $(error )
+         endif
       else
-         $(warning )
-         $(warning ######################### WARNING #########################)
-         $(warning 'USE_PREVIEW_REPO=y' set but '$(PREVIEW_REPO)' is missing. Regenerate toolkit's 'repos' directory. Remove 'USE_PREVIEW_REPO' for core builds.)
-         $(warning ######################### WARNING #########################)
-         $(warning )
+      # Try default file packaged with the toolkit...
+         ifneq ($(wildcard $(preview_repo_primary_default)),)
+            override REPO_LIST += $(preview_repo_primary_default)
+         else
+            # ...and if that fails, try the backup file in the SPECS directory.
+            ifneq ($(wildcard $(preview_repo_backup_default)),)
+            $(warning Using fallback preview repo from SPECS directory)
+               override REPO_LIST += $(preview_repo_backup_default)
+            else
+               $(warning )
+               $(warning ######################### ERROR #########################)
+               $(warning 'USE_PREVIEW_REPO=y' is set but all of the following failed:)
+               $(warning - 'PREVIEW_REPO=<path>' is not set)
+               $(warning - default '$(preview_repo_primary_default)' is missing)
+               $(warning - backup  '$(preview_repo_backup_default)' is missing)
+               $(warning ######################### ERROR #########################)
+               $(error )
+            endif
+         endif
```

Comment:
Here's the code for the idea I've mentioned in our offline discussion:
```Makefile
# Different Makefile. For instance "toolkit.mk":
preview_repo_source = $(PROJECT_ROOT)/SPECS/azurelinux-repos/azurelinux-official-preview.repo

# Make the "PREVIEW_REPO" variable "internal":
preview_repo = $(toolkit_root)/repos/mariner-official-preview.repo

ifeq ($(DAILY_BUILD_ID),)
# Leave only the file existence checks from your current `else` branch:
   ifeq ($(USE_PREVIEW_REPO),y)
      ifeq ($(wildcard $(preview_repo)),)
         ifeq ($(wildcard $(preview_repo_source)),)
           # Error message about the toolkit being neither packaged nor the user running from inside the core repository.
        endif
      endif
   endif

   override PACKAGE_URL_LIST   += https://packages.microsoft.com/azurelinux/$(RELEASE_MAJOR_ID)/preview/base/$(build_arch)
   override PACKAGE_URL_LIST   += https://packages.microsoft.com/azurelinux/$(RELEASE_MAJOR_ID)/preview/base/debuginfo/$(build_arch)
   override PACKAGE_URL_LIST   += https://packages.microsoft.com/azurelinux/$(RELEASE_MAJOR_ID)/preview/ms-oss/$(build_arch)
   override SRPM_URL_LIST      += https://packages.microsoft.com/azurelinux/$(RELEASE_MAJOR_ID)/preview/base/srpms
   override REPO_LIST          += $(preview_repo)
endif

# A native "make" approach:
$(preview_repo): $(preview_repo_source)
	@mkdir -p "$(dir $(preview_repo))
	@cp $(preview_repo_source) $(preview_repo)

# Other Makefiles for targets using "REPO_LIST":
$(graph_file): (...) $(REPO_LIST)

# Btw. we have a bunch of targets using "REPO_LIST" but not mentioning it in the prerequisites. Should we add it?
```

It assumes we make `PREVIEW_REPO` internal.

My main suggestion here is to:
1. remove one set of the manual file checking and 
2. use native `make` to do some of the work for us, since it's already centered around existence of files.

I think we can simplify a whole bunch of logic this way. The error message would also get simpler, since we'd now tell the user to either use a packaged toolkit from the core repo or run from the core itself.

Reply:
I'd have to do a bit reshuffle of the makefile to move those variables into another sub file, we do the includes after setting this stuff up.

Reply:
Actually, just moved all this into its own file